[PdrDashboard] Use polars syntax for dataframes in pdr dashboard

calina-c opened the pull request:
works on #1594.

kdetry commented on pdr_backend/pdr_dashboard/dash_components/plots.py:
```diff
@@ -345,38 +345,63 @@ def process_payouts(
+        pl.col("slot")
+        .map_elements(lambda x: UnixTimeS(int(x)).to_milliseconds(), pl.Int64)
+        .alias("slot_in_unixts"),
+    )
```
If I understood it correct, we can do it like that 

```
p = p.with_columns(
    [
        pl.lit(1).alias("predictions_crt"),
        (pl.col("payout").clip_min(0) - pl.col("stake")).alias("profit_change"),
        pl.when(pl.col("payout") > 0)
        .then(1)
        .otherwise(0)
        .alias("correct_prediction"),
        pl.col("slot")
        .cast(pl.Int64)
        .map_elements(lambda x: UnixTimeS(x).to_milliseconds(), pl.Int64).alias("slot_in_unixts")
    ]
)

p = p.with_columns(
    [
        pl.col("predictions_crt").cum_sum().alias("predictions_crt"),
        pl.col("correct_prediction").cum_sum().alias("correct_predictions_crt")
    ]
)
```

kdetry commented on pdr_backend/pdr_dashboard/dash_components/plots.py:
```diff
@@ -345,38 +345,63 @@ def process_payouts(
+        df = df.with_columns(pl.col("acc_bounds").list.get(1).alias("acc_u"))
+
+        processed.acc_l = df["acc_l"]
+        processed.acc_u = df["acc_u"]
```
We can avoid from using with_columns unnecessary

```
df = p.with_columns(
    [
        pl.struct(["correct_predictions_crt", "predictions_crt"])
        .map_elements(
            lambda x: proportion_confint(
                count=x["correct_predictions_crt"], nobs=x["predictions_crt"]
            ),
            return_dtype=pl.List(pl.Float64),
        )
        .alias("acc_bounds"),
        pl.col("acc_bounds").list.get(0).alias("acc_l"),
        pl.col("acc_bounds").list.get(1).alias("acc_u")
    ]
)

# Assigning the results to `processed` in one step
processed.acc_l = df["acc_l"]
processed.acc_u = df["acc_u"]
```

calina-c replied:
doesn't work because "acc_bounds" is only defined as part of the first "with_columns". I can't split the list in the same one. I can only do it in a second "with_columns". Or, if you have any other ideas I'm happy to try something else

kdetry replied:
you are right

kdetry commented on pdr_backend/pdr_dashboard/dash_components/plots.py:
```diff
@@ -345,38 +345,63 @@ def process_payouts(
+            return_dtype=pl.Float64,
+        )
+        .alias("accuracies")
     )
```
what is the difference with the foolowing one? I couldn't get sorry

```
p = p.with_columns(
    ((pl.col("correct_predictions_crt") / pl.col("predictions_crt")) * 100)
    .alias("accuracies")
)
```

calina-c replied:
I redid the unnecessary structs, you are absolutely right. I didn't know I could manipulate p.cols directly

kdetry commented on pdr_backend/pdr_dashboard/dash_components/plots.py:
```diff
@@ -524,43 +549,59 @@ def get_figures_and_metrics(
+
+    payouts = payouts.with_columns(
+        pl.struct(["payout", "stake"])
+        .map_elements(lambda x: max(x["payout"], 0) - x["stake"], pl.Float64)
```
why do we map here, what is the diff with the following code?


```
payouts = payouts.with_columns(
    [
        (pl.col("payout").clip_min(0) - pl.col("stake")).alias("profit"),
        pl.lit(1).alias("count"),
        pl.lit(1).cum_sum().alias("cnt_cumsum"),
        pl.when(pl.col("payout") > 0)
        .then(1)
        .otherwise(0)
        .cum_sum()
        .alias("cnt_corrpred")
    ]
)
```

kdetry commented on pdr_backend/pdr_dashboard/dash_components/plots.py:
```diff
@@ -570,37 +611,48 @@ def get_feed_figures(
-        lambda x: max(x["payout"], 0) - x["stake"], axis=1
+    payouts = payouts.with_columns(
+        pl.struct(["payout", "stake"])
+        .map_elements(lambda x: max(x["payout"], 0) - x["stake"], pl.Float64)
```
the same map question here

kdetry reviewed (changes requested):
I didn't repeatedly comment, but there may be unnecessary mappings in polar phrases. I would appreciate it if you could review them.

calina-c commented:
@kdetry I addressed your comments. I'm also thinking about adding a follow-up task for the formatting part (another issue and bullet in #1594), where we could work with cols instead of values. What do you think?

kdetry reviewed (approved):
lgtm
